main/options.c

```diff
@@ -2363,6 +2366,22 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	long patternLengthLimit;
+
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	errno = 0;
+	patternLengthLimit = strtol(parameter, NULL, 0);
+
+	if (errno != 0 || patternLengthLimit < 0)
+		error (FATAL, "-%s: Invalid pattern length limit", option);
```

Comment:
that's not really enough as it would accept `foo` just fine still, and  `12foo` too.  Correct usage is to check *both* errno and whether endptr points somewhere else than the end of the string.

```C
boolean str_to_long(const char *string, long *value, int base)
{
    char *endptr;

    errno = 0;
    *value = strtol (string, &endptr, base);
    return (*endptr) == 0 && string != endptr &&
           (errno != EINVAL && errno != ERANGE);
}

/* ... */
    long val;
    if (string_to_long(option, &val, 0) && val >= 0)
        patternLengthLimit = val;
    else /* ... */
```

or something like that